packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -74,14 +73,16 @@ export class SynthesisSession implements ISynthesisSession {
     return 'synthesize' in obj;
   }
 
-  public readonly store: ISessionStore;
+  public readonly outdir: string;
 
   private readonly artifacts: { [id: string]: cxapi.Artifact } = { };
-  private readonly buildSteps: { [id: string]: cxapi.BuildStep } = { };
   private _manifest?: cxapi.AssemblyManifest;
 
-  constructor(options: SynthesisOptions) {
-    this.store = options.store || new InMemoryStore();
+  constructor(options: SynthesisOptions = { }) {
+    this.outdir = options.outdir || fs.mkdtempSync(path.join(os.tmpdir(), 'cdk.out'));
```

Comment:
Also, no hidden dir?

Reply:
I intentionally changed to it won't be hidden because it's now not a staging cache but an actual artifact. My mental model is "gcc" which by default emits it's executable to `a.out`.

As for garbage collection - at the moment, there is none, but this is basically just for the local development use case. We could nuke it before synth every time but I think the speed benefit  of not needing to copy unchanged assets to this directory in case nothing changed outweighs the garbage accumulated for the local development case.

What do you think?

Reply:
In an ideal scenario we would (at the end of synthesis) sweep all unreferenced artifacts.

Reply:
We can always introduce this later if this becomes an issue